spec/features/admin/manage_order_detail_spec.rb

```diff
@@ -32,4 +32,46 @@
       expect(page).to have_content("successfully updated")
     end
   end
+
+  describe "canceling order details" do
+    context "canceling an item" do
+      let(:item) { create(:setup_item, facility: facility) }
+      let(:order) { create(:purchased_order, product: item) }
+      let(:order_detail) { order.order_details.first }
+
+      it "cancels the item" do
+        select "Canceled", from: "Status"
+        click_button "Save"
+
+        expect(page).to have_content("Canceled")
+        expect(page).to have_css('tfoot .currency', text: "$0.00", count: 3)
+      end
+    end
+
+    context "canceling a reservation" do
+      before do
+        instrument.price_policies.update_all(cancellation_cost: 5)
+        instrument.update_attribute(:min_cancel_hours, 1)
+      end
+
+      it "cancels without a fee" do
+        select "Canceled", from: "Status"
+        click_button "Save"
+
+        expect(page).to have_content("Canceled")
+        expect(page).to have_css('tfoot .currency', text: "$0.00", count: 3)
+      end
+
+      it "cancels with a fee" do
+        select "Canceled", from: "Status"
+        check "Add reservation cost"
+        click_button "Save"
+
+        # save_and_open_page
+
+        expect(page).to have_content("Canceled")
+        expect(page).to have_css('tfoot .currency', text: "$5.00", count: 2)
```

Comment:
@jhanggi I cannot for the life of me figure out what is wrong with the setup for this spec. Everything seems right as far as I can tell--I don't know where the fee is getting lost.

Reply:
I looked pretty quickly. It's that the `assign_price_policy` within `cancellation_fee` is returning `nil`. I don't know why yet, though. One thing to try (even on master with the require feature turned off) is canceling a New/future reservation with the checkbox checked, both in and outside the cancelation window. There may be a problem there as well.

Reply:
Ah, I figured it out. The UI is confusing in how in works in this case. There are a few situations where even if you check this box, the fee will never get applied:

* There is no `min_cancel_hours ` set on the product, even if there is a reservation cost on the price policy
* There is no reservation cost on the price policy
* The reservation is still outside the cancelation period (e.g. cancelation period of 12 hours, the reservation starts 24 hours from now). 

That last one is what's happening here. The default reservation factory is for "tomorrow at 10am". Since we're setting the `min_cancel_hours` to 1, we're still outside that window. If you change the value to 48 hours, or change the reservation_start_at, this test will pass. Or, if you make a complete reservation in the past (since the past is inherently out of the cancelation window). I'd be fine with using just about any of these scenarios. It probably would be worth explicitly setting a `reserve_start_at`/`_end_at` in the test to make the test clearer.

I don't think these are actual problems or bad logic, though the interface is confusing. If the user themself were to cancel the reservation, they would not be charged the fee, so I think it's reasonable that the admin wouldn't charge that fee either. We could probably hide the checkbox in these scenarios or show a warning next to the box, but I'm not sure it's worth the effort. We could check in with NU and see if they've had any reported confusion and if so, then we could schedule something in.

Reply:
Ahhhhhh okay. I added a comment to clarify how `min_cancel_hours` operates.